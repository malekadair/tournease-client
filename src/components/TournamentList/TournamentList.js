import React, { Component } from "react";
import Tournament from "../Tournament/Tournament";
import TournamentApiService from "../../services/tournament-api-service";
import TournamentsContext from "../../contexts/tournamentListContext";

class TournamentList extends Component {
  static contextType = TournamentsContext;
  componentDidMount() {
    this.context.clearError();
    TournamentApiService.getTournaments()
      .then(this.context.setTournaments)
      .catch(this.context.setError);
  }
  renderTournaments() {
    const { tournaments = [] } = this.context;
    return tournaments.reverse().map(tournament => (
      <Tournament key={tournament.id} tournament={tournament} />
    ));
  }

  render() {
    const { error } = this.context;
    return (
      <section className="tournamentList">
        {error ? (
          <p className="red">There was an error, try again</p>
        ) : (
            this.renderTournaments()
          )}
      </section>
    );
  }
}

export default TournamentList;
