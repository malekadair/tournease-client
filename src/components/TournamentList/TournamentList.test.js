import React from "react";
import toJson from "enzyme-to-json";
import TournamentList from "./TournamentList";

describe(`TournamentList component`, () => {
  const props = {
    id: "a",
    name: "test-class-name"
  };

  it("renders a .TournamentList by default", () => {
    const wrapper = <TournamentList />;
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("renders the TournamentList given props", () => {
    const wrapper = <TournamentList {...props} />;
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
