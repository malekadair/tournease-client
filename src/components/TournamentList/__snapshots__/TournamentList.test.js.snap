// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TournamentList component renders a .TournamentList by default 1`] = `undefined`;

exports[`TournamentList component renders the TournamentList given props 1`] = `undefined`;
